PeerioTechnologies/peerio-desktop
# feat: beacons

#### Relevant info and issue/PR links 
 https://app.clubhouse.io/peerio/story/13099/desktop-implement-area-beacon-component
https://app.clubhouse.io/peerio/story/10986/desktop-implement-spot-beacon-component-beacon-control
https://app.clubhouse.io/peerio/story/13245/desktop-implement-5-onboarding-beacons

#### Testing instructions  
The main beacon flow is intended to only show for new users immediately after signup. Also, once beacons are dismissed, their read status is saved in User's profile.

To make this easier to test, you can go to `ui-store.js` and setting `firstLogin = true`, and go to `beacon-store.ts` and comment out everything inside the `markAsRead` function. This will show you all beacons even when you're not on `firstLogin`, and it will avoid saving beacons to User profile so you can see them over and over again.

Otherwise the beacons (should) work as intended, which means most of them won't show unless you're on a completely new account on your first login after signup, and they won't reappear if you've dismissed them (or if you've taken certain other actions, e.g. navigating outside of where the beacon directs you results in all AppNav beacons being marked as read).

### Beacon sequences
Consult the Figma here for the intended flow: https://www.figma.com/file/gKlRNgEbAJGt6qNx4zrttR3B/%2311680-Desktop%3A-Initial-onboarding-Beacon-UI-placement I'll do my best to write out the flow below.

1. SpotBeacon on contacts icon in AppNav
  * After successful signup, user is directed to new Welcome screen. After 5 seconds of inactivity, beacon shows
2. AreaBeacon on user/email search in Contacts view
  * Shows if user has no contacts (whether new user or not)
3. SpotBeacon on files icon in AppNav 
  * If new user, shows after 5 seconds of user inactivity in Contacts view when viewing beacon # 2
4. AreaBeacon on "Upload" button in Files view
  * Shows if user has no files (whether new user or not)
5. SpotBeacon on chats icon in AppNav
  * If new user, shows after 5 seconds of user inactivity in Files view when viewing beacon # 4
6. AreaBeacon on plus icon on "Direct Messages" button in chat list
  * Shows if user has no chats or invites (whether new user or not)

Additional rules:
* A beacon that has been dismissed (click on the rectangle part of the beacon) will be marked as "read" and should not show again
* "New user" beacons apply only on the user's first use of the app after signing up. The `firstLogin` flag is set after a completed signup and will be reset if the user resets the app.
* If the user dismisses any of the AppNav beacons, or if the user navigates to the AppNav beacon in an order different than the one suggested by the beacons (e.g. clicking directly on chats when files is highlighted), then the AppNav beacons are all marked as read and don't show anymore.
----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
